src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -44,22 +49,39 @@ public ModelManager(ReadOnlyAddressBook addressBook, ReadOnlyUserPrefs userPrefs
 
         logger.fine("Initializing with address book: " + addressBook + " and user prefs " + userPrefs);
 
-        this.trackedAddressBook = new TrackedAddressBook(addressBook, userPrefs.getShortcutSettings());
+        this.addressBook = new AddressBook(addressBook);
         this.userPrefs = new UserPrefs(userPrefs);
-        filteredPersons = new FilteredList<>(this.trackedAddressBook.getPersonList());
+        filteredPersons = new FilteredList<>(this.addressBook.getPersonList());
         this.commandStringStash = new CommandStringStash();
         this.themeProperty = new ThemeProperty();
+        this.modelManagerStateList = new ArrayList<>();
+        this.currentStatePointer = -1;
 
         // DoConnek Pro shows all patients on startup by default.
         updateFilteredPersonList(PersonType.PATIENT.getSearchPredicate());
 
         this.selectedPerson = filteredPersons.size() == 0 ? null : filteredPersons.get(0);
+        commit();
     }
 
     public ModelManager() {
         this(new AddressBook(), new UserPrefs());
     }
 
+    /**
+     * Resets this ModelManager's data to copy another ModelManager's
+     * @param toCopyFrom ModelManager to copy from
+     */
+    public void resetData(ReadOnlyModelManager toCopyFrom) {
+        setAddressBook(toCopyFrom.addressBook);
+        setUserPrefs(toCopyFrom.userPrefs);
```

Comment:
Over here as well, we need to use the copies rather than the actual objects, since our modelManagerStateList should only have copies of the current model.

```
setUserPrefs(toCopyFrom.userPrefs.getCopy())
updateSelectedPerson(toCopyFrom.selectedPerson == null ? null : toCopyFrom.selectedPerson.getCopy())
```